wrapper/src/test/java/software/amazon/jdbc/plugin/cache/CachedResultSetTest.java

```diff
@@ -442,4 +504,283 @@ void test_get_special_date() throws SQLException {
+    assertFalse(cachedRs.wasNull());
+    assertEquals("test string", cachedRs.getNString("fieldString"));
+    assertFalse(cachedRs.wasNull());
+
+    // Test number conversion
+    assertTrue(cachedRs.next());
+    assertEquals("123", cachedRs.getNString(1));
+    assertFalse(cachedRs.wasNull());
+
+    // Test null handling
+    assertTrue(cachedRs.next());
+    assertNull(cachedRs.getNString(1));
+    assertTrue(cachedRs.wasNull());
+  }
+
+  @Test
+  void test_get_bytes() throws SQLException {
+    // Setup single column with String metadata
+    when(mockResultSet.getMetaData()).thenReturn(mockResultSetMetadata);
+    when(mockResultSetMetadata.getColumnCount()).thenReturn(1);
+    mockGetMetadataFields(1, 4);
+    // Test data
+    byte[] testBytes = {1, 2, 3, 4, 5};
+    when(mockResultSet.getObject(1)).thenReturn(testBytes, "not bytes", 123, null);
+    when(mockResultSet.next()).thenReturn(true, true, true, true, false);
+    CachedResultSet cachedRs = new CachedResultSet(mockResultSet);
+
+    // Test bytes values - both index and label versions
+    assertTrue(cachedRs.next());
+    assertArrayEquals(testBytes, cachedRs.getBytes(1));
+    assertFalse(cachedRs.wasNull());
+    assertArrayEquals(testBytes, cachedRs.getBytes("fieldByte"));
+    assertFalse(cachedRs.wasNull());
+
+    // Test non-byte array input
+    assertTrue(cachedRs.next());
+    assertArrayEquals(new byte[0], cachedRs.getBytes(1));
+    assertFalse(cachedRs.wasNull());
+
+    // Test number input
+    assertTrue(cachedRs.next());
+    assertArrayEquals(new byte[0], cachedRs.getBytes(1));
+    assertFalse(cachedRs.wasNull());
+
+    // Test null handling
+    assertTrue(cachedRs.next());
+    assertNull(cachedRs.getBytes(1));
+    assertTrue(cachedRs.wasNull());
+  }
+
+  @Test
+  void test_get_boolean() throws SQLException {
+    // Setup single column with String metadata
+    when(mockResultSet.getMetaData()).thenReturn(mockResultSetMetadata);
+    when(mockResultSetMetadata.getColumnCount()).thenReturn(1);
+    mockGetMetadataFields(1, 3);
+    // Test data: boolean, numbers, strings, null
+    when(mockResultSet.getObject(1)).thenReturn(
+        true, false, 0, 1, -5, "true", "false", "invalid", null);
+    when(mockResultSet.next()).thenReturn(true, true, true, true, true, true, true, true, true, false);
+
+    CachedResultSet cachedRs = new CachedResultSet(mockResultSet);
+
+    // Test actual boolean values - both index and label versions
+    assertTrue(cachedRs.next());
+    assertTrue(cachedRs.getBoolean(1));
+    assertFalse(cachedRs.wasNull());
+    assertTrue(cachedRs.getBoolean("fieldBoolean"));
+
+    assertTrue(cachedRs.next());
+    assertFalse(cachedRs.getBoolean(1));
+    assertFalse(cachedRs.wasNull());
+
+    // Test number conversions: 0 = true, non-zero = false
+    assertTrue(cachedRs.next());
+    assertTrue(cachedRs.getBoolean(1)); // 0 → true
+
+    assertTrue(cachedRs.next());
+    assertFalse(cachedRs.getBoolean(1)); // 1 → false
```

Comment:
`0` means false and `1` means true? 

https://docs.oracle.com/javase/8/docs/api/java/sql/ResultSet.html#getBoolean-int-